src/EditorFeatures/CSharpTest/Completion/CompletionProviders/AwaitCompletionProviderTests.cs

```diff
@@ -279,5 +279,231 @@ public async Task TestAwaitInLock_TopLevel()
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnValueTask()
+        {
+            var valueTaskAssembly = typeof(ValueTask).Assembly.Location;
+            await VerifyItemExistsAsync(@$"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"" LanguageVersion=""{LanguageVersion.CSharp9.ToDisplayString()}"">
+        <MetadataReference>{valueTaskAssembly}</MetadataReference>
+        <Document FilePath=""Test2.cs"">
+using System.Threading.Tasks;
+
+class C
+{{
+  async Task F(ValueTask someTask)
+  {{
+    someTask.$$
+  }}
+}}
+        </Document>
+    </Project>
+</Workspace>
+", "await");
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotOnCustomAwaitable()
+        {
+            await VerifyKeywordAsync(@"
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+public class DummyAwaiter: INotifyCompletion {
+    public bool IsCompleted => true;
+    public void OnCompleted(Action continuation) => continuation();
+    public void GetResult() {}
+}
+
+public class CustomAwaitable
+{
+    public DummyAwaiter GetAwaiter() => new DummyAwaiter();
+}
+
+static class Program
+{
+    static async Task Main()
+    {
+        var awaitable = new CustomAwaitable();
+        awaitable.$$;
+    }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotDot()
+        {
+            await VerifyKeywordAsync(@"
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main(Task someTask)
+    {
+        someTask.$$.;
+    }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotBeforeType()
+        {
+            await VerifyKeywordAsync(@"
+using System;
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main(Task someTask)
+    {
+        someTask.$$
+        Int32 i = 0;
+    }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitSuggestAfterDotBeforeAnotherAwait()
+        {
+            await VerifyKeywordAsync(@"
+using System;
+using System.Threading.Tasks;
+
+static class Program
+{
+    static async Task Main(Task someTask)
+    {
+        someTask.$$
+        await Test();
+    }
+
+    async Task Test() { }
+}", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotAfterDotOnTaskIfAlreadyAwaited()
+        {
+            await VerifyAbsenceAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    await someTask.$$
+  }
+}
+", LanguageVersion.CSharp9);
+        }
+
+        [Fact]
+        public async Task TestDotAwaitNotInLock()
+        {
+            await VerifyAbsenceAsync(@"
+using System.Threading.Tasks;
+
+class C
+{
+  async Task F(Task someTask)
+  {
+    lock(this) { someTask.$$ }
```

Comment:
interewsting.  where was the code that prevented this?  i didn't see it.

Reply:
It's in IsDotAwaitContext. The last thing I check is, if IsAwaitKeywordContext is true at the insertion position.

Reply:
ah ok.  thanks.